common/app/model/dotcomrendering/pageElements/PageElement.scala

```diff
@@ -867,27 +867,40 @@ object PageElement {
         val imageAssets = element.assets.zipWithIndex
           .map { case (a, i) => ImageAsset.make(a, i) }
 
-        val imageSources: Seq[ImageSource] = DotcomRenderingImageRoleWidthByBreakpointMapping.all.map {
-          case (weighting, widths) =>
-            val srcSet: Seq[SrcSet] = widths.breakpoints.flatMap { b =>
-              Seq(
-                ImgSrc.srcsetForBreakpoint(
-                  b,
-                  DotcomRenderingImageRoleWidthByBreakpointMapping.immersive.breakpoints,
-                  maybeImageMedia = Some(ImageMedia(imageAssets)),
-                ),
-                ImgSrc.srcsetForBreakpoint(
-                  b,
-                  DotcomRenderingImageRoleWidthByBreakpointMapping.immersive.breakpoints,
-                  maybeImageMedia = Some(ImageMedia(imageAssets)),
-                  hidpi = true,
-                ),
-              )
-            }.flatten
-            // A few very old articles use non-https hosts, which won't render
-            val httpsSrcSet = srcSet.map(set => set.copy(src = ensureHTTPS(set.src)))
-            ImageSource(weighting, httpsSrcSet)
-        }.toSeq
+        def getImageBreakpointMapping(
+            isMainMedia: Boolean,
+            isImmersive: Boolean,
+        ): ImageRoleWidthsByBreakpointMapping = {
+          if (isMainMedia) MainMedia
+          else if (isImmersive) ImmersiveMedia
+          else BodyMedia
```

Comment:
This is - I believe - correct. (It's also really easy to read!) But the naming for these mappings really confused me so I'm adding a comment for others and also for future us.

Essentially, the issue is an overload for the term 'immersive'. This work both describes a type of article but it is also a weighting

`MainMedia` is the size mappings for all main media images based on weighting
`ImmersiveMedia`  is for all **body** images on an **immersive** article
`BodyMedia` is all body images for articles that are _not_ immersive